### PR TITLE
Ensure WooPay ‘Enabled by Default’ Value Is Correctly Set In Sandbox Mode

### LINKED ISSUE #9421: WooPay is not automatically enabled in sandbox test-drive flow
### Describe the bug

When onboarding in sandbox mode via the test-drive flow (see https://github.com/Automattic/woocommerce-payments/pull/9109), WooPay is not automatically enabled.

Originally reported in https://github.com/Automattic/woopay/issues/2922

### To Reproduce

1. Spin up a new JN site

> [!WARNING]
> Do not use the dev tools option when creating the site. See p1726100013463429/1726097935.447679-slack-C022WMN88KG for more details.

2. Install and activate WooCommerce
3. Configure the store location as a US state (e.g., California)
4. Install and activate WooPayments
5. Onboard in sandbox mode
6. Navigate to **Payments > Settings**
7. Note that WooPay is not automatically enabled

### Actual behavior

WooPay is not automatically enabled when in sandbox mode. It is automatically enabled on a live account.

We've been able to reproduce in 8.1.1 and 8.2.0.

### Expected behavior

WooPay is automatically enabled in sandbox mode, which creates parity with the live mode experience.

### PR DESCRIPTION
**Pull request by lovo-h**
Fixes #9421 

#### Changes proposed in this Pull Request

The changes in this PR ensure that the WooPay 'enabled by default' value is correctly set in the KYC flow's sandbox mode. Previously, it was only set in live mode.

#### Testing instructions

> [!WARNING]
> The `WCPay Dev` plugin should not be used for the following tests since it impacts the results.

**Regression Test: Ensure WooPay is enabled by default in live mode**
* Create a new merchant site.
* Ensure [sandbox mode](https://woocommerce.com/document/woopayments/testing-and-troubleshooting/sandbox-mode/) is disabled.
* As a merchant, navigate to the `Payments` and click the `Connect your store` button.
* Go through the KYC flow to completion.
* Navigate to `Payments > Settings` and scroll down to the `Express checkouts` section.
* Ensure `WooPay` checkbox is checked.

**Ensure WooPay is enabled by default in sandbox mode**
* Create a new merchant site.
* Ensure [sandbox mode](https://woocommerce.com/document/woopayments/testing-and-troubleshooting/sandbox-mode/) is enabled.
* As a merchant, navigate to the `Payments` and click the `Connect your store` button.
* Go through the KYC flow to completion.
* Navigate to `Payments > Settings` and scroll down to the `Express checkouts` section.
* Ensure `WooPay` checkbox is checked.

**Ensure unit tests pass**
* Ensure all tests from `npm run test:php` run successfully.
* Ensure all tests from `npm run test:js` run successfully.

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [x] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `9898` or branch name `fix/9421-auto-enable-woopay-in-sandbox-mode` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=fix/9421-auto-enable-woopay-in-sandbox-mode) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: e4ee3fd43a7997a8e3ef30bb0d7a82370f88e069
- Build time: 2024-12-12 16:52:36 UTC

_Note: the build is updated when a new commit is pushed to this PR._

File: includes/class-wc-payments-account.php [hunk 1]
```diff
@@ -2004,6 +2004,9 @@ private function init_stripe_onboarding( string $setup_mode, string $wcpay_conne
 			$collect_payout_requirements
 		);
 
+		// Store the 'woopay_enabled_by_default' flag in a transient to be used later.
+		set_transient( self::WOOPAY_ENABLED_BY_DEFAULT_TRANSIENT, filter_var( $onboarding_data['woopay_enabled_by_default'] ?? false, FILTER_VALIDATE_BOOLEAN ), DAY_IN_SECONDS );
+
```
Comment (asumaran) [inline includes/class-wc-payments-account.php]: What was the reason for relocating this here?

**Review by asumaran — COMMENTED** [review]
The code looks good. I was able to reproduce the bug and confirm the fix in sandbox mode. However, I couldn’t verify it in live mode since it requires a verified US Stripe account. I'm holding off on approving the PR for now.

File: includes/class-wc-payments-account.php (hunk 1 above)
Comment (lovo-h) [reply includes/class-wc-payments-account.php]: If [this if-then condition](https://github.com/Automattic/woocommerce-payments/pull/9898/files/8bf19e73c7d978eb4351a9259a2fafa3ddbf7519#diff-1cf209a1776d0208a1a6125d0ab5d3958b8631909cea0bce44b22f922eb1bdbfL2009) resolves to `true`, we would have never reached the `set_transient( ... )` call. This is what was occurring in the sandbox mode: this if-then condition was resolving to `true`.

**Review by bborman22 — APPROVED** [review]
I tested this several times with a Pressable site in live mode. In two cases it worked as expected and WooPay was enabled by default when I completed the onboarding process. In the third case it didn't work, but there was also issues with completing the onboarding flow from the Stripe side (verification of site data given the fact it was a test site).

I also reviewed the code changes and nothing seemed to be a gap in the flow that would prevent enablement, so I think we are good here in avoiding any regressions.

LGTM!

**Review by asumaran — APPROVED** [review]
I can confirm that, even with the recent changes, WooPay remains enabled as expected in sandbox mode after the onboarding.

**Review by bborman22 — APPROVED** [review]
I was able to confirm that with the latest updates, WooPay was enabled by default when onboarding with a live account on a Pressable site!

LGTM!

**Pull request merged**